### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Andrey1104**
Знімок екрана (53)
Знімок екрана (54)
Знімок екрана (55)
Знімок екрана (56)
Знімок екрана (57)
Знімок екрана (58)
Знімок екрана (59)
Знімок екрана (60)
Знімок екрана (61)
Знімок екрана (62)
Знімок екрана (63)

### CODE REVIEW

File: taxi/urls.py [hunk 1]
```diff
@@ -44,7 +50,39 @@
     path("cars/<int:pk>/delete/", CarDeleteView.as_view(), name="car-delete"),
     path("drivers/", DriverListView.as_view(), name="driver-list"),
     path(
-        "drivers/<int:pk>/", DriverDetailView.as_view(), name="driver-detail"
+        "drivers/<int:pk>/",
+        DriverDetailView.as_view(),
+        name="driver-detail"
+    ),
+    path(
+        "drivers/create",
```
Comment (vasylhnatiuk) [inline taxi/urls.py]: add
```suggestion
        "drivers/create/",
```

Comment (vasylhnatiuk) [reply taxi/urls.py]: add / in the end of path everywhere

File: taxi/views.py [hunk 2]
```diff
@@ -87,3 +89,63 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
...
+    form_class = DriverLicenseUpdateForm
+    template_name = "taxi/driver_license_update.html"
+
+
+class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+def driver_license_delete(request: HttpRequest, pk: int) -> HttpResponse:
```
Comment (vasylhnatiuk) [inline taxi/views.py]: rewrite it as a class view

File: taxi/views.py [hunk 3]
```diff
@@ -87,3 +89,63 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
...
+    driver = Driver.objects.get(pk=pk)
+    return render(
+        request,
+        "taxi/driver_confirm_license_delete.html",
+        {"driver": driver}
+    )
+
+
+def driver_confirm_delete(
```
Comment (vasylhnatiuk) [inline taxi/views.py]: rewrite it as a class view

File: taxi/views.py [hunk 4]
```diff
@@ -87,3 +89,63 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
...
+    driver.save()
+    return HttpResponseRedirect(
+        reverse_lazy(
+            "taxi:driver-detail",
+            args=[driver.id]
+        )
+    )
+
+
+def update_driver_to_car_view(
```
Comment (vasylhnatiuk) [inline taxi/views.py]: rewrite it as a class view

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: taxi/forms.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+    class Meta:
+        model = Driver
+        fields = ("license_number", )
+
+    def clean_license_number(self) -> None:
+        license_number = self.cleaned_data["license_number"]
+        print(license_number)
```
Comment (pavlejviki) [inline taxi/forms.py]: remove print

File: taxi/urls.py [hunk 6]
```diff
@@ -55,28 +54,23 @@
...
     ),
     path(
-        "drivers/<int:pk>/license_delete",
-        driver_license_delete,
-        name="driver-license-delete"
-    ),
-    path(
-        "drivers/<int:pk>/confirm_license_delete",
-        driver_confirm_delete,
+        "drivers/<int:pk>/confirm_license_delete/",
```
Comment (pavlejviki) [inline taxi/urls.py:67]: you should add a driver deletion page not a license deletion page.

File: taxi/views.py [hunk 7]
```diff
@@ -107,45 +107,46 @@ class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
...
+
+
+class UpdateDriverCarView(LoginRequiredMixin, generic.UpdateView):
+    def get(
+            self,
+            request: HttpRequest,
+            *args,
+            **kwargs
+    ) -> HttpResponseRedirect:
+        pk_d = self.kwargs.get("pk_d")
```
Comment (pavlejviki) [inline taxi/views.py]: use more descriptive variable names

**Review by pavlejviki — CHANGES_REQUESTED** [review]

File: taxi/urls.py (hunk 6 above)
Comment (Andrey1104) [reply taxi/urls.py:67]: its additional option. I immediately misunderstood condition)) then decided to keep this option in project. driver_deleter  also in the project

Comment (Andrey1104) [reply taxi/urls.py:67]: If you go to your profile there will be a button "delete driver" , if you go to someone else’s profile there will be no button

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

**Pull request closed without merging**